metaserver/src/main/scala/scala/meta/languageserver/CompilerConfig.scala

```diff
@@ -0,0 +1,94 @@
+package scala.meta.languageserver
+
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.util.Properties
+import com.typesafe.scalalogging.LazyLogging
+import org.langmeta.io.AbsolutePath
+import org.langmeta.io.Classpath
+
+/**
+ * Configuration to load up a presentation compiler.
+ *
+ * In sbt, one compiler config typically corresponds to one project+config.
+ * For example one sbt project with test/main/it configurations has three
+ * CompilerConfig.
+ *
+ * @param sources list of source files for this project
+ * @param scalacOptions space separated list of flags to pass to the Scala compiler
+ * @param dependencyClasspath File.pathSeparated list of *.jar and classDirectories.
+ *                  Includes both dependencyClasspath and classDirectory.
+ * @param classDirectory The output directory where *.class files are emitted
+ *                       for this project.
+ * @param sourceJars File.pathSeparated list of *-sources.jar from the
+ *                   dependencyClasspath.
+ */
+case class CompilerConfig(
+    sources: List[AbsolutePath],
+    scalacOptions: List[String],
+    classDirectory: AbsolutePath,
+    dependencyClasspath: List[AbsolutePath],
+    sourceJars: List[AbsolutePath]
+) {
+  override def toString: String =
+    s"CompilerConfig(" +
```

Comment:
since we have pprint as a dependency, you can probably just do `pprint.stringify(this)`